wezterm-ssh/src/lib.rs

```diff
@@ -9,3 +9,8 @@ pub use config::*;
 pub use host::*;
 pub use pty::*;
 pub use session::*;
+
+// NOTE: Re-exported as is exposed in a public API of this crate
+pub use filedescriptor::FileDescriptor;
+pub use portable_pty::Child;
+pub use ssh2::{FileStat, OpenFlags, OpenType, RenameFlags};
```

Comment:
One of the things that I'd like to be possible in the near future is allowing for wezterm-ssh to be built against `libssh` instead of `libssh2` via a crate feature (libssh has better and broader support for different ssh key types), so exposing the raw ssh2 types directly would be undesirable.

Could you make wrapper types for these that avoid exposing the ssh2 types directly?
I know that's a bit of a boilerplatey PITA!

As part of that, I think it would be good to keep an eye on `libssh`'s API to try to avoid committing us to an API that cannot be represented if we're using that under the covers.  I think it should be mostly fine as both libraries are coupled to the ssh2 protocol.

And while thinking API style thoughts: the ssh2 crate exposes libssh2's terminology. I think it would be good to make a pass over the names of things and tweak them to more closely match Rust's terminology for filesystem operations.

Reply:
Yeah, I can make wrapper types. To my knowledge, sftp was only introduced in ssh 2, meaning that it wouldn't be available in libssh, right? Just a heads up that distant would require ssh2 and couldn't use libssh.

Would it make sense to make a public module called ssh2 or something that wezterm exposes instead with those types? That way it could be optionally included with a feature and would be clear what is exposed:

```rust
#[cfg(feature = "ssh2")]
pub mod ssh2 {
    pub use ::ssh2::FileStat;
    pub use ::ssh2::OpenFlags;
    pub use ::ssh2::OpenType;
    pub use ::ssh2::RenameFlags;
}
```

Or did you want wrapper types regardless? I can do either, just checking.

Reply:
I know it's more work, but I'd prefer wrapper types so that it is easier to evolve the API (and naming!) for different ssh backends.

`libssh` supports sftp: https://api.libssh.org/master/libssh_tutor_sftp.html

(`libssh` support SSH protocol 2!)

Reply:
> (`libssh` support SSH protocol 2!)

Oh nice! I'll whip up some wrapper types, then.

Reply:
Created wrapper types. Things like `OpenType`, `FileStat`, `OpenFlags`, and `RenameFlags` all exist in libssh as well with same values, so wrapper type mirroring those values (mostly) should be enough.